pkg/restapi/webhook/webhook_test.go

```diff
@@ -154,10 +182,78 @@ func TestWebhookNotificationClient500Response(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	err := sendNotification(fmt.Sprintf("http://%s%s", clientHost, clientHandlerPattern))
+	err := notify(fmt.Sprintf("http://%s%s", clientHost, clientHandlerPattern), nil)
 	require.Contains(t, err.Error(), "500 Internal Server Error", err.Error())
 }
 
+func getTestBasicMessageJSON() []byte {
+	return []byte(`
+   {
+       "connection_id": "SomeConnectionID",
+       "message_id": "SomeMessageId",
+		"content": "SomeContent",
+		"state": "SomeState"
+   }
+		`)
+}
+
+func runClient(testClientData clientData) error {
+	err := listenAndStopAfterReceivingNotification(testClientData.clientHost)
+	if err != nil {
+		return err
+	}
+	close(testClientData.subscriberReceivedNotification)
+	return nil
+}
+
+func listenAndStopAfterReceivingNotification(addr string) error {
+	m := http.NewServeMux()
+	srv := &http.Server{Addr: addr, Handler: m}
+	ctx, cancel := context.WithCancel(context.Background())
+	m.HandleFunc("/", func(resp http.ResponseWriter, req *http.Request) {
```

Comment:
Are you sure your tests are working?
You didn't write SUCCESS code to response here and tests are still passing

Reply:
Handlers default to returning 200 OK if you don't write explicitly write a header/message. Let me know if you still think there might be something wrong.